### PR TITLE
timepicker sets the dirty flag via writeValue

### PR DESCRIPTION
**Pull request by washtubs**
Basically the issue is that if you give the timepicker an initial value or you set it manually (as opposed to the user interacting with the element directly), it will set the dirty flag to true. This happens as a part of the `onChange` function. The dirty flag should stay false until a user interacts with the element. So only those functions should result in calling `onChange` i.e.: `changeHours`, `changeMinutes`, `changeSeconds`, `updateHours`, `updateMinutes`, `updateSeconds`, `toggleMeridian`. 

In general `writeValue` should never call `onChange`, which is essentially what was happening here.

By the way, my company is stuck on 3.x at this time. If this change is acceptable, is it also possible to get a patch for 3.x?

Demo on [stackblitz](https://stackblitz.com/edit/angular-ngx-bootstrap-timepicker-dirty).

### CODE REVIEW

**Review by daniloff200 — APPROVED** [review]

**Comment by valorkin** [comment]
@washtubs hey, yes you can do PR to 3.x tag, and referer here
mentioning me will help to do it faster

File: src/timepicker/timepicker.component.ts [hunk 1]
```diff
@@ -177,7 +178,6 @@ export class TimepickerComponent
       .subscribe((value: Date) => {
         // update UI values if date changed
         this._renderTime(value);
-        this.onChange(value);
```
Comment (valorkin) [inline src/timepicker/timepicker.component.ts:183]: there are a chance that without mark for check it will not work

**Review by valorkin — COMMENTED** [review]
@IraErshova could you please check my comment

**Comment by daniloff200** [comment]
@washtubs  is it still actual change for you?

**Comment by daniloff200** [comment]
Well, 11 months passed, and there's still no response. Also, almost 2 years since it has been opened. I think I can close it for now, but, you always can ping me to reopen it back.

**Pull request closed without merging**